p2p/subscriber.go

```diff
@@ -92,6 +105,7 @@ func (s *Subscriber[H]) SetVerifier(val func(context.Context, H) error) error {
 			log.Errorw("unmarshalling header",
 				"from", p.ShortString(),
 				"err", err)
+			s.metrics.reject(ctx)
```

Comment:
Optiona / Alternative approach:
Could be just a validation wrapper func with single switch statement. It could wrap `pval` to observe validation result.

Reply:
I don't get it

Reply:
func observeResutl(validationResult) validationResult
Example: https://github.com/celestiaorg/celestia-node/blob/2c5eef712ded6f5b383990d55df73da5d143aada/share/p2p/peers/metrics.go#L209

Reply:
I think this switch should live in our business logic, not inside of metrics